File: 加密保护/ASProtect SKE v2.56 Build 03.17/Examples/Reg Trial/BCB/regkey.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include <windows.h>
#include "regkey.h"
#include "include\aspr_api.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TfrmRegKey *frmRegKey;
char* LocalHardwareID = NULL;
//---------------------------------------------------------------------------
__fastcall TfrmRegKey::TfrmRegKey(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmRegKey::FormCreate(TObject *Sender)
{
  LocalHardwareID = GetHardwareID();
  Edit1->Text     = StrPas( LocalHardwareID );
}
//---------------------------------------------------------------------------

void __fastcall TfrmRegKey::Button4Click(TObject *Sender)
{
  char buffer[256];
  memset( (char*)buffer, 0, sizeof(buffer));
  lstrcat(buffer,"mailto:support@aspack.com?subject=Hardware%20ID%20for%20Advanced%20Application%201.0&body=Hardware%20ID is%20:%20");
  lstrcat(buffer,LocalHardwareID);
  lstrcat(buffer,"%0d%0aRegistration%20information%20:");
  ShellExecute(GetDesktopWindow(), "open", buffer, NULL, NULL, SW_SHOWNORMAL);
}
//---------------------------------------------------------------------------

void __fastcall TfrmRegKey::btnRegOkClick(TObject *Sender)
{

  if (CheckKeyAndDecrypt( eCode->Text.c_str(), eName->Text.c_str(), TRUE ) == TRUE)
  {
    MessageBox(Handle, "Thank you for your registration!", "Registration", MB_ICONINFORMATION);
    Close();
  } else
    MessageBox(Handle, "Key is not valid, please contact manufacture!","Registration", MB_ICONWARNING);

}
//---------------------------------------------------------------------------

void __fastcall TfrmRegKey::btnRegPasteClick(TObject *Sender)
{
  eCode->Clear();
  eCode->PasteFromClipboard();
}
//---------------------------------------------------------------------------

File: 加密保护/ASProtect SKE v2.56 Build 03.17/Examples/Reg Trial/Delphi/DFM/regkey.pas
{$J+}
unit regkey;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls;

type
  TfrmRegKey = class(TForm)
    lCode: TLabel;
    btnRegOk: TButton;
    btnRegCancel: TButton;
    btnRegPaste: TButton;
    Button4: TButton;
    Edit1: TEdit;
    Label2: TLabel;
    eCode: TEdit;
    lName: TLabel;
    eName: TEdit;
    procedure btnRegPasteClick(Sender: TObject);
    procedure Button4Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure btnRegOkClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmRegKey: TfrmRegKey;

implementation

uses
  ShellAPI,
  aspr_api;

var
  LocalHardwareID : PChar;

{$R *.DFM}

//------------------------------------------------------------------------------

procedure TfrmRegKey.btnRegPasteClick(Sender: TObject);
begin
  eCode.Clear;
  eCode.PasteFromClipboard;
end;

//------------------------------------------------------------------------------

procedure TfrmRegKey.Button4Click(Sender: TObject);
begin
  ShellExecute(GetDesktopWindow(), 'open', PChar('mailto:support@aspack.com?subject=Hardware%20key%20for%20Advanced%20Application%201.0&body=Hardware%20key is%20:%20'+Edit1.Text+
                                                 '%0d%0aName%20for%20registration%20:'), nil, nil, SW_SHOWNORMAL);
end;

//------------------------------------------------------------------------------

procedure TfrmRegKey.FormCreate(Sender: TObject);
begin
  LocalHardwareID := GetHardwareID;
  Edit1.Text      := StrPas( LocalHardwareID );
end;

//------------------------------------------------------------------------------

procedure TfrmRegKey.btnRegOkClick(Sender: TObject);
begin

  If CheckKeyAndDecrypt( PChar(eCode.Text), PChar(eName.Text), True ) then
  begin
    MessageBox(Handle, 'Thank you for your registration!', 'Registration', MB_ICONINFORMATION);
    Close;
  end else
    MessageBox(Handle, 'Key is not valid, please contact manufacture!','Registration', MB_ICONWARNING);

end;

//------------------------------------------------------------------------------

end.